lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -635,23 +635,14 @@ private ColumnIndexValueConverter()
 
         private Function<ByteBuffer, Object> getConverter(PrimitiveType primitiveType)
         {
-            switch (primitiveType.getPrimitiveTypeName()) {
-                case BOOLEAN:
-                    return buffer -> buffer.get(0) != 0;
-                case INT32:
-                    return buffer -> buffer.order(LITTLE_ENDIAN).getInt(0);
-                case INT64:
-                    return buffer -> buffer.order(LITTLE_ENDIAN).getLong(0);
-                case FLOAT:
-                    return buffer -> buffer.order(LITTLE_ENDIAN).getFloat(0);
-                case DOUBLE:
-                    return buffer -> buffer.order(LITTLE_ENDIAN).getDouble(0);
-                case FIXED_LEN_BYTE_ARRAY:
-                case BINARY:
-                case INT96:
-                default:
-                    return buffer -> Binary.fromReusedByteBuffer(buffer);
-            }
+            return switch (primitiveType.getPrimitiveTypeName()) {
```

Comment:
nit: I'd squash the two commits with modern switches

Reply:
I had to drop those commits as they were causing the error prone plugin to crash due to some bug of it's own